enumerables.rb

```diff
@@ -0,0 +1,157 @@
+# rubocop:disable Style/CaseEquality, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(to_a[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    i = 0
+    while i < length
+      arr << self[i] if yield(self[i])
+      i += 1
+    end
+    arr
+  end
+
+  def my_all?(aaa = nil)
+    if aaa
+
+      my_each { |elem| return false unless aaa === elem }
+
+    end
+    unless block_given?
+      i = 0
+      while i < length
+        return false unless self[i]
+
+        i += 1
+      end
+      return true
+    end
+    bool = true
+    i = 0
+    while i < length
+      bool = false unless yield(self[i])
+      i += 1
+    end
+    bool
+  end
+end
+
+module Enumerable
+  def my_any?(aaa = nil)
+    if aaa
+
+      my_each { |elem| return true if aaa === elem }
+    end
+    unless block_given?
+      i = 0
+      while i < length
+        return true unless self[i]
+
+        i += 1
+      end
+      return false
+    end
+    bool = false
+    i = 0
+    while i < length
+      bool = true if yield(self[i])
+      i += 1
+    end
+    bool
+  end
+
+  def my_none?(aaa = nil, &prc)
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
array = Array.new(100) { rand(0...9) }
array.my_none?(String) == true #true
```
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
words = %w[dog door rod blade]
words.my_none?(5) == words.none?(5) #true
```
- [ ] Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
 words.my_none?(/z/) == words.none?(/z/) #true
 words.my_none?(/d/) == words.none?(/d/) #true
```
- [ ] when no block or argument is given
```ruby
true_array = [nil, false, true, []] 
false_array = [nil, false, nil, false]
false_array.my_none? == true #true
true_array.my_none? == false #false
```